lib/kb_trimmomatic/kb_trimmomaticImpl.py

```diff
@@ -810,14 +845,20 @@ def execTrimmomatic(self, ctx, input_params):
                                       #'info':
                                           })
                 if some_unpaired_fwd_output_created:
-                    output_readsSet_obj = { 'description': input_readsSet_obj['data']['description']+" Trimmomatic unpaired fwd reads",
+                    output_readsSet_obj = { 'description': str(input_readsSet_obj['data']['description'])+" Trimmomatic unpaired fwd reads",
                                             'items': items
                                             }
                     output_readsSet_name = str(input_params['output_reads_name'])+'_trimm_unpaired_fwd'
                     unpaired_fwd_readsSet_ref = setAPI_Client.save_reads_set_v1 ({'workspace_name': input_params['output_ws'],
                                                                                   'output_object_name': output_readsSet_name,
                                                                                   'data': output_readsSet_obj
                                                                                   })['set_ref']
+                    unpaired_fwd_SampleSet_ref = self._save_RNASeqSampleSet(
+                                                            items,
+                                                            input_params['output_ws'],
+                                                            output_readsSet_name + '_SampleSet',
+                                                            reads_desc_ext,
+                                                            single_reads)
```

Comment:
I don't think this is right. If an input paired-end RNASeqSampleSet is input, only the paired-end samples should be saved in the output RNASeqSampleSet object, the unpaired reads shouldn't be saved in the object.

If the input RNASeqSampleSet was single-end, then a single-end RNASeqSampleSet should be saved. Presumably all the output files are unpaired.

Reply:
It's in a if-else block. It's called only when input object is a set object.